pkg/expression/generator/builtin_threadsafe.go

```diff
@@ -0,0 +1,222 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
```

Comment:
you can add some build tags like other files:

https://github.com/pingcap/tidb/blob/aa83e4e0d5fbba3aa502abd99bbcddc5f85f1e5a/pkg/expression/generator/compare_vec.go#L14-L17